talk/objectorientation/objectsclasses.tex

```diff
@@ -49,16 +52,46 @@
       // let's square a
       myObj.squareA();
     \end{cppcode*}
-    \columnbreak
+      \onslide<2>
+    \begin{cppcode*}{gobble=2}
+      class MyFirstClass {
+        int a;
+        public:
+        void squareA() {
+          a *= a;
+        }
+        int sum(int b) {
+          return a + b;
+        }
+      };
+
+      MyFirstClass myObj;
+      myObj.a = 2; // error
+
+      // let's square a
+      myObj.squareA();
+    \end{cppcode*}
+    \end{overprint}
+      \columnbreak
     \center
-    \null \vfill
     \begin{tikzpicture}
       \classbox{MyFirstClass}{
         int a; \\
         void squareA(); \\
         int sum(int b);
       }
     \end{tikzpicture}
+    \pause
+    \null \vfill
+    {
+      \setlength\textwidth\columnwidth
+    \begin{block}{struct vs.\ class}
+      \begin{itemize}
+        \item In structs, every member is accessible
+        \item In classes, members are ``private'' unless made public 
```

Comment:
```suggestion
        \item In structs, every member and base is ``public''
        \item In classes, they are ``private'' unless made public 
```

Reply:
We are just before the slide where we deal with inheritance here. I put this box to not confuse people why we say class al they time, but the keyword is `struct`.

Reply:
Alright, makes sense!